benchmarks/persistence/persistence.py

```diff
@@ -0,0 +1,229 @@
+"""Classes for persisting benchmark results."""
+
+import json
+import os
+from typing import Protocol
+
+import boto3
+import boto3.session
+from attr import define, field
+from attrs.validators import instance_of
+from git import Repo
+
+from benchmarks import Benchmark, Result
+
+
+class PersistenceObjectInterface(Protocol):
+    """Interface for constructing the path of a result object."""
+
+    def get_full_file_path(self) -> str:
+        """Construct the path of a result object.
+
+        Returns:
+            The path of the result object.
+        """
+
+    def get_object(self) -> dict:
+        """Construct the object to be persisted.
+
+        Returns:
+            The object to be persisted.
+        """
+
+
+class PersistenceHandlingInterface(Protocol):
+    """Interface for persisting experiment results."""
+
+    def write_object(self, object: PersistenceObjectInterface) -> None:
+        """Store the result of a benchmark.
+
+        Args:
+            object: The object to be persisted.
+        """
```

Comment:
Now that the writer is separated, my original question again pops up: why can't we use a framework like `fsspec` for this?

Reply:
@AdrianSosic I've implemented what we discussed, please let me know what you think. I do believe that it is a good direction we agreed upon :)

Reply:
I think that can be resolved.